### DISCUSSION TITLE
webpage

### DISCUSSION THREAD

**Discussion by constantinosergiou** [spatie/laravel-health#37] [Q&A] [answered]
i have added it in routes and a get

local.ERROR: Invalid route action: [App\Http\Controllers\Spatie\Health\Http\Controllers\HealthCheckResultsController]

**Comment by cshawaus** [comment, marked as answer]
I too experienced the same problem but found a simple workaround for now until the real reason is discovered.

```php
Route::get('health', [HealthCheckResultsController::class, '__invoke']);
```

**Reply by constantinosergiou** [reply]
@cshawaus  thank you

**Reply by jeversen** [reply]
Thanks! I've got the same problem here and using your workaround for now!

**Comment by trippo** [comment]
The problem was the base namespace of the routes:


```php
// App\Providers\RouteServiceProvider
Route::middleware('web')
                ->namespace($this->namespace)
                ->group(base_path('routes/web.php'));
```
That adds the base namespace before HealthCheckResultsController FQN

So you can remove `->namespace` line and update all old routes definition or you can add this

```php
Route::get('health', '\\'.HealthCheckResultsController::class);
```

**Comment by yuxxeun** [comment]
> I too experienced the same problem but found a simple workaround for now until the real reason is discovered.
> 
> ```php
> Route::get('health', [HealthCheckResultsController::class, '__invoke']);
> ```

thanks man, you saved my day. 
i was stuck at this point for ~5 hours, lol.

**Comment by delino12** [comment]
I think testing packages should have a JD on it's own. Got stuck for 3 hours +
